src/v4l/v4lt.rs

```diff
@@ -0,0 +1,148 @@
+
+pub(crate) struct V4lDevice<'a> {
+    device: &'a mut Device,
+}
+
+pub(crate) struct V4ltOutputs<'b, 'c> {
+    vidsrc: &'b mut MmapStream<'b>,
+    video_format: Option<StreamFormat>,
+    device: &'c mut Device,
+}
+
+// The stream from the camera will be using one of these formats
+//
+// This is used as part of `StreamOutput` to give hints about
+// the format of the stream
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+enum StreamFormat {
+    // H264 (AVC) video format
+    H264,
+    // H265 (HEVC) video format
+    H265,
+}
+
+impl<'b, 'c> StreamOutput for V4ltOutputs<'b, 'c> {
+    fn write(&mut self, media: BcMedia) -> StreamOutputError {
+        match media {
+            BcMedia::Iframe(payload) => {
+                let video_type = match payload.video_type {
+                    VideoType::H264 => StreamFormat::H264,
+                    VideoType::H265 => StreamFormat::H265,
+                };
+                self.set_format(Some(video_type));
+                //self.vidsrc.write_all(&payload.data)?;
+
+                let (buf_out, buf_out_meta) = OutputStream::next(self.vidsrc).unwrap();
+
+                // Output devices generally cannot know the exact size of the output buffers for
+                // compressed formats (e.g. MJPG). They will however allocate a size that is always
+                // large enough to hold images of the format in question. We know how big a buffer we need
+                // since we control the input buffer - so just enforce that size on the output buffer.
+                let buf_out = &mut buf_out[0..payload.data.len()];
+
+                buf_out.copy_from_slice(&payload.data);
+                buf_out_meta.field = 0;
+                //buf_out_meta.bytesused = buf_in_meta.bytesused;
+            }
+            BcMedia::Pframe(payload) => {
+                let video_type = match payload.video_type {
+                    VideoType::H264 => StreamFormat::H264,
+                    VideoType::H265 => StreamFormat::H265,
+                };
+                self.set_format(Some(video_type));
+                //self.vidsrc.write_all(&payload.data)?;
+
+                let (buf_out, buf_out_meta) = OutputStream::next(self.vidsrc).unwrap();
+
+                // Output devices generally cannot know the exact size of the output buffers for
+                // compressed formats (e.g. MJPG). They will however allocate a size that is always
+                // large enough to hold images of the format in question. We know how big a buffer we need
+                // since we control the input buffer - so just enforce that size on the output buffer.
+                let buf_out = &mut buf_out[0..payload.data.len()];
+
+                buf_out.copy_from_slice(&payload.data);
+                buf_out_meta.field = 0;
+                //buf_out_meta.bytesused = buf_in_meta.bytesused;
+            }
+            _ => {
+                //Ignore other BcMedia
+            }
```

Comment:
You may want to also listen to the `BcMediaInfoV2` and `BcMediaInfoV1` packets. They contain information such as the width and height. Check out `crates/core/src/bcmedia/model.rs`. These messages are only sent once at the beginning of a stream so you'll have to cache the results in your `V4ltOutputs`